From 1d9b35e9d10839a9d513eed551034fd02e855f92 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Mon, 12 Oct 2015 11:46:43 +0800
Subject: [PATCH] video: rockchip: hdmi: fix parsing 1st block edid

If parsing 1st block edid failed, we think it is good
only when there is more than one external block.

Change-Id: I1eb11405dade41ebb2ad041ba909de0c7ea120f8
Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/video/rockchip/hdmi/rockchip-hdmi-edid.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/hdmi/rockchip-hdmi-edid.c b/drivers/video/rockchip/hdmi/rockchip-hdmi-edid.c
index 610937e703ed..1ade35096198 100644
--- a/drivers/video/rockchip/hdmi/rockchip-hdmi-edid.c
+++ b/drivers/video/rockchip/hdmi/rockchip-hdmi-edid.c
@@ -122,7 +122,8 @@ int hdmi_edid_parse_base(unsigned char *buf,
 	fb_edid_to_monspecs(buf, pedid->specs);
 
 out:
-	if (rc != E_HDMI_EDID_SUCCESS && *extend_num > 4)
+	if ((rc != E_HDMI_EDID_SUCCESS) &&
+	    (*extend_num < 1 && *extend_num > 4))
 		return rc;
 	else
 		return E_HDMI_EDID_SUCCESS;
-- 
2.35.3

